src/components/ui/Link/README.md

```diff
@@ -4,13 +4,62 @@ A `<Link>` provides a way of navigating to other content, either on the same sit
 import { Paragraph } from '@octopusthink/nautilus';
 
 <Paragraph>
-  Fill out the <Link to="/sign-up">sign up form</Link> to create your account.
+  Fill out the <Link href="#sign-up">sign up form</Link> to create your account.
 </Paragraph>
 ```
 
 ## Usage
 
-Use a `<Link>` component where you would usually use an `<a>` tag in HTML. If you use the `href` prop, this link will behave exactly like an `<a>` tag; if you use the `to` prop instead, this component will use `react-router-dom`'s `Link` component to output a link that your app can use to navigate to other routes you have defined.
+Use a `<Link>` component where you would usually use an `<a>` tag in HTML.
+
+By default, Nautilus will output an `<a>` tag for every `Link` component. If you want to use a custom `Link` component throughout your app you can use the `LinkComponent` prop in your `Nautilus` higher-order component. Here's how you would use various link components from popular router libraries:
+
+```jsx
+import Nautilus, { Paragraph, Tabs } from '@octopusthink/nautilus';
+import { Link as ReachRouterLink } from '@reach/router';
+import { createMemoryHistory } from 'history';
+import { Link as ReactRouterLink, Router } from 'react-router-dom';
+
+<Tabs>
+  <Tabs.Tab label="Reach Router Link">
+    <Nautilus config={{LinkComponent: ReachRouterLink}}>
+      <Paragraph>
+        This link uses <Link to="#react-router">Reach Router</Link> for links.
+      </Paragraph>
+    </Nautilus>
+  </Tabs.Tab>
+
+  <Tabs.Tab label="React Router Link">
+    <Nautilus config={{LinkComponent: ReactRouterLink}}>
+      <Router history={createMemoryHistory()}>
+        <Paragraph>
+          But this link uses <Link to="#react-router">React Router</Link>.
+        </Paragraph>
+      </Router>
+    </Nautilus>
+  </Tabs.Tab>
+</Tabs>
+```
+
+If you want to override a specific `Link` component, you can use the `asComponent` prop:
+
+```jsx
+import Nautilus, { Paragraph } from '@octopusthink/nautilus';
+import { createMemoryHistory } from 'history';
+import { Link as ReactRouterLink, Router } from 'react-router-dom';
+
+<Nautilus>
+  <Router history={createMemoryHistory()}>
+    <Paragraph>
+      Fill out the
+      <Link asComponent={ReactRouterLink} to="/sign-up">sign up form</Link> to
```

Comment:
Other systems seem to use `as`. Would it make sense here for us to follow this pattern?

Reply:
I'm definitely cool with that. In fact: we might even get it for free as we use the `styled()` HOC on our components. I'll have a look!